cmd/ttn-lw-cli/commands/use.go

```
@@ -0,0 +1,140 @@
+	fileName := base
+	if user {
+		dir, err := os.UserConfigDir()
+		if err != nil {
+			return "", err
+		}
+		fileName = filepath.Join(dir, base)
+	}
+
+	_, err := os.Stat(fileName)
+	if err == nil && !overwrite {
+		logger.Warnf("%s already exists. Use --overwrite", fileName)
+		return "", errFileExists.WithAttributes("file", fileName)
+	}
+
+	return fileName, nil
+}
+
+var (
+	errNoHost      = errors.DefineInvalidArgument("no_host", "no host set")
+	errFileExists  = errors.DefineAlreadyExists("file_exists", "`{file}` exists")
+	errFailWrite   = errors.DefinePermissionDenied("fail_write", "failed to write `{file}`")
+	configFileName = ".ttn-lw-cli.yml"
+	caFileName     = "ca.pem"
+
+	useCommand = &cobra.Command{
+		Use:               "use [host]",
+		Short:             "Use",
+		PersistentPreRunE: preRun(),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			insecure, _ := cmd.Flags().GetBool("insecure")
+			ca, _ := cmd.Flags().GetBool("ca")
+			user, _ := cmd.Flags().GetBool("user")
+			overwrite, _ := cmd.Flags().GetBool("overwrite")
+			credentialsID, _ := cmd.Flags().GetString("credentials-id")
+
+			if len(args) != 1 {
+				return errNoHost
+			}
+
+			host := args[0]
+			conf := MakeDefaultConfig(host, insecure)
+			if credentialsID != "" {
+				conf.CredentialsID = credentialsID
+			}
+
+			// Get CA certificate from server
+			if !insecure && ca {
+				logger.Infof("Will retrieve certificate from %s", conf.NetworkServerGRPCAddress)
+				caFile, err := destPath("ca.pem", user, overwrite)
+				if err != nil {
+					return err
+				}
+
+				conn, err := tls.Dial("tcp", conf.NetworkServerGRPCAddress, &tls.Config{InsecureSkipVerify: true})
+				if err != nil {
+					return err
+				}
+				defer conn.Close()
+
+				var b bytes.Buffer
+				w := bufio.NewWriter(&b)
+				for _, cert := range conn.ConnectionState().PeerCertificates {
+					if !cert.IsCA {
+						continue
+					}
+					pem.Encode(w, &pem.Block{Type: "CERTIFICATE", Bytes: cert.Raw})
```

Do not discard errors

---

Printing a warning if `pem.Encode` fails